R/zchunk_LA154.Transport_china.R

```diff
@@ -0,0 +1,196 @@
+#' @return Depends on \code{command}: either a vector of required inputs,
+#' a vector of output names, or (if \code{command} is "MAKE") all
+#' the generated outputs: \code{L154.in_EJ_province_trn_m_sz_tech_F}, \code{L154.out_mpkm_province_trn_nonmotor_Yh}, \code{L154.in_EJ_province_trn_F}. The corresponding file in the
+#' original data system was \code{LA154.Transport.R} (gcam-china level1).
+#' @details Transportation energy data was downscaled in proportion to NBS province-level transportation energy data
+#' @details Transportation nonmotor data was downscaled in proportion to province population
+#' @importFrom assertthat assert_that
+#' @importFrom dplyr filter mutate select
+#' @importFrom tidyr gather spread
+#' @author AJS June 2017
+module_gcam.china_LA154.Transport <- function(command, ...) {
+  if(command == driver.DECLARE_INPUTS) {
+    return(c(FILE = "gcam-china/trnUCD_NBS_mapping",
+             "L154.in_EJ_R_trn_m_sz_tech_F_Yh",
+             "L154.out_mpkm_R_trn_nonmotor_Yh",
+             "L100.Pop_thous_province",
+             "L101.inNBS_Mtce_province_S_F",
+             "L101.NBS_use_all_Mtce"))
+  } else if(command == driver.DECLARE_OUTPUTS) {
+    return(c("L154.in_EJ_province_trn_m_sz_tech_F",
+             "L154.out_mpkm_province_trn_nonmotor_Yh",
+             "L154.in_EJ_province_trn_F"))
+  } else if(command == driver.MAKE) {
+
+    all_data <- list(...)[[1]]
+
+    # Load required inputs
+    trnUCD_NBS_mapping <- get_data(all_data, "gcam-china/trnUCD_NBS_mapping")
+    L154.in_EJ_R_trn_m_sz_tech_F_Yh <- get_data(all_data, "L154.in_EJ_R_trn_m_sz_tech_F_Yh")
+    L154.out_mpkm_R_trn_nonmotor_Yh <- get_data(all_data, "L154.out_mpkm_R_trn_nonmotor_Yh")
+    L100.Pop_thous_province <- get_data(all_data, "L100.Pop_thous_province")
+    L101.inNBS_Mtce_province_S_F <- get_data(all_data, "L101.inNBS_Mtce_province_S_F")
+    L101.NBS_use_all_Mtce <- get_data(all_data, "L101.NBS_use_all_Mtce")
+    # ===================================================
+
+      # Silence package notes
+      GCAM_region_ID <- year <- value <- UCD_sector <- size.class <- UCD_technology <- UCD_fuel <- fuel <- EBProcess <- EBMaterial <-
+      fuel_sector <- province <- sector <- value_national <- value_share <- pop <- value_mode <- NULL
+
+      # Calculate the province-wise percentages for each of NBS's sector/fuel combinations that is relevant for disaggregating
+      # nation-level transportation energy to the provinces
+
+      # This starting table is transportation energy consumption by GCAM region (and other variables)
+      # We will first subset this data for only the china and values that are > 0 in the historical periods
+
+      L154.in_EJ_R_trn_m_sz_tech_F_Yh %>%
+        filter(year %in% HISTORICAL_YEARS, GCAM_region_ID == 11) %>% # Filter for the china and for historical years only
+        filter(value != 0) %>% # Here any rows with value of 0 will be lost, even if other years of the same group are nonzero
```

Comment:
According to the comments, it only works in the wide format - in the long format, it only drops the years with zero value.